python/ray/tune/function_runner.py

```diff
@@ -476,34 +477,37 @@ def detect_checkpoint_function(train_func, abort=False):
     return validated
 
 
-def wrap_function(train_func):
+def wrap_function(train_func, warn=True):
     if hasattr(train_func, "__mixins__"):
         inherit_from = train_func.__mixins__ + (FunctionRunner, )
     else:
         inherit_from = (FunctionRunner, )
 
+    func_args = inspect.getfullargspec(train_func).args
+    use_checkpoint = detect_checkpoint_function(train_func)
+    if len(func_args) > 1:  # more arguments than just the config
+        if "reporter" not in func_args and not use_checkpoint:
+            raise ValueError(
+                "Unknown argument found in the Trainable function. "
+                "Arguments other than the 'config' arg must be one "
+                "of ['reporter', 'checkpoint_dir']. Found: {}".format(
+                    func_args))
+
+    use_reporter = "reporter" in func_args
+    if not use_checkpoint and not use_reporter:
```

Comment:
Why do we check for `use_reporter` here? Shouldn't `not use_checkpoint` be enough for the warning?

As an aside, I don't think there's enough documentation on `reporter`. The only thing I can find is `Object passed into your function that you can report status through.`.  What's the difference between this and `tune.report` and `CLIReporter`?

Reply:
`reporter` is deprecated API. We should make that clear wherever it is offered (could you link where you found that?)

Reply:
https://docs.ray.io/en/master/tune/api_docs/trainable.html#statusreporter, https://docs.ray.io/en/master/tune/user-guide.html#handling-large-datasets